libs/agno/agno/api/team.py

```
@@ -1,34 +1,71 @@
-from typing import Dict, List, Optional
-
-from httpx import Response
-
-from agno.api.api import api, invalid_response
+from agno.api.api import api
 from agno.api.routes import ApiRoutes
-from agno.api.schemas.team import TeamSchema
-from agno.api.schemas.user import UserSchema
+from agno.api.schemas.team import TeamRunCreate, TeamSessionCreate
+from agno.cli.settings import agno_cli_settings
 from agno.utils.log import logger
 
 
-def get_teams_for_user(user: UserSchema) -> Optional[List[TeamSchema]]:
-    logger.debug("--**-- Reading teams for user")
+def create_team_session(session: TeamSessionCreate, monitor: bool = False) -> None:
+    if not agno_cli_settings.api_enabled:
+        return
+
+    logger.debug("--**-- Logging Team Session")
+    with api.AuthenticatedClient() as api_client:
+        try:
+            response = api_client.post(
+                ApiRoutes.TEAM_SESSION_CREATE if monitor else ApiRoutes.TEAM_TELEMETRY_SESSION_CREATE,
+                json={"session": session.model_dump(exclude_none=True)},
+            )
+            response.raise_for_status()
```

1. i dont thing `raise_for_status` is supported on httpx
2. even if it is, we shouldnt throw errors if telemetry fails - fail silently and continue

---

Yes, good catch.

---

Well it does get caught by the exception handler? And httpx does support raise for status. I tested this and it works for example at the moment where the API is not done yet, otherwise no exception is raised, no debug log is made